### PR TITLE
Documentation: warn user that testsuite will modify local environment

### PR DESCRIPTION
**Pull request by aalemayhu**
Closes: #1671 (Documentation: Add warning that running tests/ will modify system state)
Signed-off-by: Alexander Alemayhu <alexander@alemayhu.com>

### CODE REVIEW

File: Documentation/contributing.rst [hunk 1]
```diff
@@ -213,6 +213,17 @@ commands, respectively:
 Testsuite
 ~~~~~~~~~
 
+.. Warning::
+
+  Running the testsuite will modify the host environment. If you are using the
+  default VM that might not be an issue, but if you are running bare metal or a
+  different VM the tests might fail or worst case remove possibly important
+  configuration. Specifically, they are modifying tables and flushing chains,
```
Comment (joestringer) [inline Documentation/contributing.rst]: Are these IPTables tables / chains?

File: Documentation/contributing.rst [hunk 2]
```diff
@@ -213,6 +213,17 @@ commands, respectively:
...
+  kernel configurations change for IP. Also adding and removing networking
```
Comment (joestringer) [inline Documentation/contributing.rst]: Does `kernel configurations change for IP` mean assigning IP addresses to devices, or did you have other things in mind?

**Review by joestringer — COMMENTED** [review]
This looks useful, although it'd be nice if it's stated in a way that provides a little more clarity to developers about how they might know something was modified. For instance, at the moment it refers to 'tables' and 'chains', which I guess are done through iptables (and not, for instance, nftables); and kernel configurations. One way to look at it is that the host environment changes include stuff like:
* IPtables configuration
* TC configuration
* Network devices
* Routing?
* etc.

File: Documentation/contributing.rst (hunk 1 above)
Comment (aalemayhu) [reply Documentation/contributing.rst]: Yes

File: Documentation/contributing.rst (hunk 2 above)
Comment (aalemayhu) [reply Documentation/contributing.rst]: Was thinking about `sysctl` but there other things in the setup script
https://github.com/cilium/cilium/blob/master/bpf/init.sh#L74-L78

File: Documentation/contributing.rst (hunk 1 above)
Comment (joestringer) [reply Documentation/contributing.rst]: OK, it might be good to name-drop iptables in this paragraph.

Comment (aalemayhu) [reply Documentation/contributing.rst]: fixed

File: Documentation/contributing.rst (hunk 2 above)
Comment (aalemayhu) [reply Documentation/contributing.rst]: drop this part or reword it?

Comment (joestringer) [reply Documentation/contributing.rst]: Maybe just reword it to specify for example, networking options via ``sysctl``?

File: Documentation/contributing.rst [hunk 3]
```diff
@@ -210,9 +210,22 @@ commands, respectively:
...
 Testsuite
 ~~~~~~~~~
 
+.. Warning::
+
+  Running the testsuite will modify the host environment. If you are using the
+  default VM that might not be an issue, but if you are running bare-metal or a
+  different VM the tests might fail or worst case remove possibly important
+  configuration. Specifically, they are modifying tables and flushing chains
+  via ``iptables``, kernel configurations change for IP via ``sysctl``. Also
```
Comment (tgraf) [inline Documentation/contributing.rst]: Looks great, this one sentence should be reworded to:

    Specifically, they are modifying the state and configuration of system including,
    changes to ``iptables`` configuration, kernel configuration via ``sysctl``, adding
    and removing networking devices, routes, etc. via ``iproute2``.

**Review by tgraf — CHANGES_REQUESTED** [review]

**Review by tgraf — APPROVED** [review]

**Review by joestringer — APPROVED** [review]
